### PR TITLE
Add builtin logic to `CairoRunner.Initialize`

### PR DESCRIPTION
**Pull request by fmoletta**
Depends on and contains changes from #14 & #29
Integrates builtin logic into `CairoRunner.Initialize()` & `NewCairoRunner`
Adds example on how to add a builtin in `NewCairoRunner`
Adds method `Memory.ValidateExistingMemory`

### CODE REVIEW

File: pkg/runners/cairo_runner_test.go [hunk 1]
```diff
@@ -10,13 +10,28 @@ import (
 	"github.com/lambdaclass/cairo-vm.go/pkg/vm/memory"
 )
 
+func TestNewCairoRunnerInvalidBuiltin(t *testing.T) {
+	// Create a Program with one fake instruction
+	program_data := make([]memory.MaybeRelocatable, 1)
+	empty_identifiers := make(map[string]parser.Identifier, 0)
+	program_data[0] = *memory.NewMaybeRelocatableFelt(lambdaworks.FeltFromUint64(uint64(1)))
```
Comment (gabrielbosio) [inline pkg/runners/cairo_runner_test.go]:
```suggestion
	program_data[0] = *memory.NewMaybeRelocatableFelt(lambdaworks.FeltOne())
```

File: pkg/vm/memory/memory_test.go [hunk 2]
```diff
@@ -276,3 +276,47 @@ func TestMemorySegmentsLoadDataTwoElements(t *testing.T) {
...
 }
+
+func TestMemoryValidateExistingMemoryOk(t *testing.T) {
+	mem_manager := memory.NewMemorySegmentManager()
+	mem_manager.AddSegment()
+	mem := &mem_manager.Memory
+	// Load Values to memory
+	for i := uint(0); i < 15; i++ {
+		key := memory.NewRelocatable(0, i)
+		val := memory.NewMaybeRelocatableFelt(lambdaworks.FeltFromUint64(uint64(i)))
```
Comment (gabrielbosio) [inline pkg/vm/memory/memory_test.go:287]:
```suggestion
		val := memory.NewMaybeRelocatableFelt(lambdaworks.FeltOne())
```

File: pkg/vm/memory/memory_test.go [hunk 3]
```diff
@@ -276,3 +276,47 @@ func TestMemorySegmentsLoadDataTwoElements(t *testing.T) {
...
+}
+
+func TestMemoryValidateExistingMemoryErr(t *testing.T) {
+	mem_manager := memory.NewMemorySegmentManager()
+	mem_manager.AddSegment()
+	mem := &mem_manager.Memory
+	// Load Values to memory
+	for i := uint(0); i < 15; i++ {
+		key := memory.NewRelocatable(0, i)
+		val := memory.NewMaybeRelocatableFelt(lambdaworks.FeltFromUint64(uint64(i)))
```
Comment (gabrielbosio) [inline pkg/vm/memory/memory_test.go:309]:
```suggestion
		val := memory.NewMaybeRelocatableFelt(lambdaworks.FeltOne())
```

**Review by gabrielbosio — APPROVED** [review]

**Comment by gabrielbosio** [comment]
Looks good, left some minor comments

File: pkg/vm/memory/memory_test.go (hunk 3 above)
Comment (fmoletta) [reply pkg/vm/memory/memory_test.go:309]: It says i

File: pkg/vm/memory/memory_test.go (hunk 2 above)
Comment (fmoletta) [reply pkg/vm/memory/memory_test.go:287]: It says i

**Review by toni-calvin — APPROVED** [review]
Looks good! 👍

File: pkg/vm/memory/memory_test.go (hunk 3 above)
Comment (gabrielbosio) [reply pkg/vm/memory/memory_test.go:309]: ups

**Pull request merged**